### PR TITLE
doc: Initial documentation for PyKokkos (Sphinx format)

### PR DESCRIPTION
**Pull request by gliga**
This PR introduces documentation pages for PyKokkos.

The documentation covers the key concepts: patterns, policies, work units. Also, it covers the docker instructions.

Current limitations: there are still policies that are not described, as well as execution spaces. Will be a separate PR. There will also be a separate PR to make this into GitHub pages.

### CODE REVIEW

File: docs/patterns.rst [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+following signature:
+
+.. code-block:: python
+
+   parallel_for([label], policy, workunit, [keyword arguments])
+
+* **label** is an optional string value helpful for debugging and
+  profiling
+* **policy** specifies the way computations are executed (execution
+  place and number of workunits to run in parallel). In its simplies
```
Comment (NaderAlAwar) [inline docs/patterns.rst:29]: simplies -> simplest

File: docs/policies.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
+
+Policies
+========
+
+Policy is the second key concept in PyKokkos (:doc:`concepts`).  Each
+pattern (:doc:`patterns`) accept a policy as an argument. A policy
```
Comment (NaderAlAwar) [inline docs/policies.rst:6]: accept -> accepts

**Review by NaderAlAwar — CHANGES_REQUESTED** [review]
Small spelling fixes needed

File: docs/patterns.rst (hunk 1 above)
Comment (gliga) [reply docs/patterns.rst]: done

File: docs/policies.rst (hunk 2 above)
Comment (gliga) [reply docs/policies.rst]: done

**Review by NaderAlAwar — APPROVED** [review]

**Pull request merged**